docs/apps/components/number.md

```diff
@@ -0,0 +1,193 @@
+
+4. **Data Validation**: For example, in a form collecting age, validate that the input is a positive integer within a reasonable age range.
+
+5. **Visual Enhancements**: For example, create custom input fields with color changes or animations based on input values or provide interactive sliders or spinners for precise value adjustments.
+
+## API
+
+### Events
+
+| **Name**| **Description**|
+|---------|----------------|
+| `onChange (element)`| The onChange event occurs when the value of a field is changed|
+| `onFocus (element)`| The onfocus event occurs when a field gets focus|
+| `onBlur (element)`| The onblur event occurs when a field loses focus|
+
+### Methods
+
+| **Name**| **Description**|**Parameters**|
+|---------|----------------|--------------|
+|`this.addClassName`|Add style class to a form item|`(fields: String`&#124;`String[], className: String)`|
+|`this.removeClassName`|Remove style class from a form item|`(fields: String`&#124;`String[], className: String)`|
+|`this.getComponent`|Returns a component whose id has been passed as a parameter|`(component_ID: String) : Object)`|
+|`this.getValues`|Gets the current values of all fields|`() : Object`|
+|`this.hide`|Hides the field|`(fields: String`&#124;`String[])`|
+|`this.show`|Displays the field|`(fields: String`&#124;`String[])`|
+|`this.disable`| Disable number field from user interaction|`(fields: String`&#124;`String[])`|
+|`this.enable`| Enable number field from user interaction|`(fields: String`&#124;`String[])`|
+|`this.getValue`|Get A Value From a component|`(fieldName: String)`|
+|`this.setData`|Set the data in the field|`(Value: Object)`|
+
+!!! Info
+    1. The show() and hide() methods can also be used to control the visibility of an alert in response to user input.
+    2. **Fields** refers to a **component ID** or a **list of component IDs**. You can fetch the ID from the **Component Attribute** panel in the Page Builder.
+    3. Before using **this.show(fields)**, make sure the component is **hidden**. This can be done using **this.hide(fields)** or by **enabling** the **Hidden checkbox** in the Component Attribute panel.
+    4. Before using **this.enable(fields)**, make sure the component is **disabled**. This can be done using **this.disable(fields)** or by **enabling** the **Disabled checkbox** in the Component Attribute panel.
+
+#### Steps to use the methods for the Page Builder components
+
+1. Go to Form Attribute :material-menu-right: Action Panel :material-menu-right: Setting :material-menu-right: (Mounted | refresh | click 'Add action').
+2. Write the method/code as shown in the **Example** below.
+3. Click on `Save`.
+4. On the main screen click on `Save` again.
+5. Click on `Preview` to see the code in action.
+<img src= "/apps/components/img/check1.png">
```

Comment:
Please use an image that uses the numberId instead of checkbox

Reply:
done